pkg/scheduler/scheduler.go

```diff
@@ -780,11 +781,29 @@ type podConditionUpdaterImpl struct {
 
 func (p *podConditionUpdaterImpl) update(pod *v1.Pod, condition *v1.PodCondition) error {
 	klog.V(3).Infof("Updating pod condition for %s/%s to (%s==%s, Reason=%s)", pod.Namespace, pod.Name, condition.Type, condition.Status, condition.Reason)
-	if podutil.UpdatePodCondition(&pod.Status, condition) {
-		_, err := p.Client.CoreV1().Pods(pod.Namespace).UpdateStatus(context.TODO(), pod, metav1.UpdateOptions{})
-		return err
-	}
-	return nil
+	// First try to update using the pod that was passed in. If conflict occurs, get latest pod and try again.
+	updatePodConditionBackoff := wait.Backoff{
+		Steps:    5,
+		Duration: 100 * time.Millisecond,
+		Jitter:   1.0,
+	}
```

Comment:
nit: move this as one level up to be a local variable, so as to be shared with other functions.

Reply:
I moved it up